### PR TITLE
Updating Jobbergate Documentation

### PR DESCRIPTION
**Pull request by dusktreader**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,85 +1,61 @@
...
+     <img
+       src="https://img.shields.io/pypi/pyversions/jobbergate?label=python-versions&logo=python&style=plastic"
+       alt="python versions"
+     />
+     <img
+       src="https://img.shields.io/pypi/v/jobbergate?label=pypi-version&logo=python&style=plastic"
+       alt="pypi version"
+     />
+     <img
+       src="https://img.shields.io/pypi/l/jobbergate?style=plastic"
```
Comment (fschuch) [inline README.rst:31]: Jobbergate itself is not at pypi, we need some sub-project here to make it work.

```suggestion
       src="https://img.shields.io/pypi/l/jobbergate-cli?style=plastic"
```

File: jobbergate-docs/src/apps/jobbergate_api/usage.rst [hunk 2]
```diff
@@ -0,0 +1,123 @@
+======================
+ Using Jobbergate API
+======================
+
+The Jobbergate API is intended to be consumed by automated processes or other applications. These include
+the Jobbergate CLI and Jobbergate Agent, but it may serve any number of other applications.
+
+The Jobbergate API is a standard RESTful API that uses HTTP for commuinication.
```
Comment (fschuch) [inline jobbergate-docs/src/apps/jobbergate_api/usage.rst:8]:
```suggestion
The Jobbergate API is a standard RESTful API that uses HTTP for communication.
```

File: jobbergate-docs/src/developer_guide/s3_file_storage.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Besides the information kept in the database, application and job-scripts files are
+stored in an S3 Bucket.
+
+The bucket is named according to the ``S3_BUCKET_NAME`` environment variable used by
+the Jobbergate API.
+
+Within the the bucket, there should be two top-level folders, and the structure should
+appear similar to:
+
+.. code-block::
```
Comment (fschuch) [inline jobbergate-docs/src/developer_guide/s3_file_storage.rst:14]: This is outdated. The supporting files changed the file structure. I'll create a task to cover this.

File: jobbergate-docs/src/tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,565 @@
+==========
+ Tutorial
```
Comment (fschuch) [inline jobbergate-docs/src/tutorial.rst:2]: Really nice job on this one

File: jobbergate-docs/src/developer_guide/integration_testing.rst [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+* Logging out through the CLI
+
+
+Setup
+-----
+
+First, start up the Jobbergate system in docker-compose. Navigate to the
+``jobbergate-composed`` subproject and run the following command:
+
+   .. code-block:: console
```
Comment (fschuch) [inline jobbergate-docs/src/developer_guide/integration_testing.rst:36]: No need for indentation on this one, it is leading to unexpected empty blocks. Please, change all the other occurrences on this page.

```suggestion
.. code-block:: console
```

File: jobbergate-docs/src/index.rst [hunk 6]
```diff
@@ -5,14 +5,32 @@
...
+   Developer Guide <developer_guide/index>
+   Attribution <attribution>
+
+
+TODO
+----
+
+* Write tutorial using docker-compose
+* Write a CI doc
+* Write "Setting up keycloak" guide
```
[the remark this answers is not in the document]
Comment (dusktreader) [reply jobbergate-docs/src/index.rst:36]: Didn't get around to writing a "setting up keycloak" guide, but:
* out of time for this task
* not really within the scope of jobbergate docs

File: README.rst (hunk 1 above)
Comment (dusktreader) [reply README.rst]: Removed the whole block in commit d2ca89e.

File: README.rst [hunk 7]
```diff
@@ -1,85 +1,61 @@
+.. raw:: html
+
+   <p align="center">
+     <img
+       src="https://img.shields.io/github/workflow/status/omnivector-solutions/jobbergate/test_on_push/main?label=main-build&logo=github&style=plastic"
```
[the remark this answers is not in the document]
Comment (dusktreader) [reply README.rst]: Fixed in commit d2ca89e.

File: README.rst [hunk 8]
```diff
@@ -1,85 +1,61 @@
...
+enable the re-use and remote submission of job scripts to a Slurm cluster.
 
+There are 3 main components of Jobbergate that are hosted as sub-projects of this
+repository:
 
-jobbergate-cli
---------------
+* `jobbergate-api <https://github.com/omnivector-solutions/jobbergate/jobbergate-api>`_
+* `jobbergate-cli <https://github.com/omnivector-solutions/jobbergate/jobbergate-cli>`_
+* `jobbergate-agent <https://github.com/omnivector-solutions/jobbergate/jobbergate-agent>`_
```
[the remark this answers is not in the document]
Comment (dusktreader) [reply README.rst]: Removed in commit d2ca89e.

File: jobbergate-cli/README.rst [hunk 9]
```diff
@@ -2,88 +2,28 @@
...
-    jobbergate --help
-
-.. note::
-
-   It is possible to use raw sbatch parameters in create-job-script
-
-   Use the `--sbatch-params` multiple times to use as many parameters as needed in the
-   following format ``--sbatch-params='-N 10'`` or
-   ``--sbatch-params='--comment=some_comment'``
+   jobberggate job-script --help
```
[the remark this answers is not in the document]
Comment (dusktreader) [reply jobbergate-cli/README.rst]: Fixed in commit d2ca89e.

File: jobbergate-cli/README.rst [hunk 10]
```diff
@@ -2,88 +2,28 @@
...
-So, for example, to create a minor release, you would run:
-
-.. code-block:: console
-
-   make relase-minor
-
-You must have permission to push commits to the main branch to create a release.
-
-If the release script fails, contact a maintainer to debug and fix the release.
+   jobberggate job-script list --help
```
[the remark this answers is not in the document]
Comment (dusktreader) [reply jobbergate-cli/README.rst]: Fixed in commit d2ca89e.

File: jobbergate-cli/tests/subapps/job_scripts/test_app.py [hunk 11]
```diff
@@ -368,3 +369,30 @@ def test_delete__makes_request_and_sends_terminal_message(
...
+
+def test_show_files__success(
+    respx_mock,
+    make_test_app,
+    dummy_job_script_data,
+    dummy_job_script_files,
+    dummy_domain,
+    cli_runner,
+    mocker,
+):
```
[the remark this answers is not in the document]
Comment (dusktreader) [reply jobbergate-cli/tests/subapps/job_scripts/test_app.py:382]: Fixed in commit d2ca89e.

File: jobbergate-docs/src/apps/jobbergate_api/usage.rst (hunk 2 above)
Comment (dusktreader) [reply jobbergate-docs/src/apps/jobbergate_api/usage.rst]: Fixed in commit d2ca89e.

File: jobbergate-docs/src/index.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (dusktreader) [reply jobbergate-docs/src/index.rst]: Removed in commit d2ca89e.

File: jobbergate-docs/src/developer_guide/integration_testing.rst (hunk 5 above)
Comment (dusktreader) [reply jobbergate-docs/src/developer_guide/integration_testing.rst]: Embarassing! Fixed in commit d2ca89e.

**Review by fschuch — APPROVED** [review]

**Pull request merged**